datagateway_api/common/database/models.py

```diff
@@ -170,7 +153,7 @@ def get_related_entity(self, entity):
         :return: The entity
         """
         try:
-            return getattr(self, entity)
+            return getattr(self, entity if entity[-1] == "s" else entity.upper())
```

Comment:
Is this testing for "pluralness" - it works fine for now, but perhaps a comment explaining this would be helpful in case an new ICAT entity type is created which doesn't "pluralise" with an "s" on the end.

Reply:
Updated the docstring of this function to give some detail about the logic behind the plural stuff.